processor/src/decorators/mod.rs

```diff
@@ -148,6 +153,44 @@ impl Process {
 
         Ok(())
     }
+
+    /// Given a quadratic extension field element ( say a ) on stack top, this routine computes
+    /// multiplicative inverse of that element ( say b ) s.t.
+    ///
+    /// a * b = 1 ( mod P ) | b = a ^ -1, P = irreducible polynomial x^2 - x + 2 over F_q, q = 2^64 - 2^32 + 1
+    ///
+    /// Input on stack expected in following order
+    ///
+    /// [coeff_1, coeff_0, ...]
+    ///
+    /// While computed multiplicative inverse is put on advice provider in following order
+    ///
+    /// [coeff'_0, coeff'_1, ...]
+    ///
+    /// Meaning when a Miden program is going to read it from advice tape, it'll see
+    /// coefficient_0 first and then coefficient_1.
+    ///
+    /// Note, in case input operand is zero, division by zero error is returned, because
+    /// that's a non-invertible element of extension field.
+    fn inject_ext2_inv_result(&mut self) -> Result<(), ExecutionError> {
+        let coef0 = self.stack.get(1);
+        let coef1 = self.stack.get(0);
+
+        let elm = Ext2Element::new(coef0, coef1);
+        if elm == Ext2Element::ZERO {
+            return Err(ExecutionError::DivideByZero(self.system.clk()));
+        }
+
+        let inv_elm = elm.inv();
+
+        let elm_arr = [inv_elm];
+        let coeffs = Ext2Element::as_base_elements(&elm_arr);
```

Comment:
Side note: unfortunately, we have to do it like this. It would be very convenient if `QuadExtension` struct exposed a method to get base elements directly.

Reply:
Yes, would have been better.